src/allocation_engine/README.md

```diff
@@ -0,0 +1,135 @@
+# Interval Tree allocation engine
+
+## Design
+
+This interval tree implementation is defined to work as an allocation
+engine for memory-mapped I/O and port I/O address space. It provides
+underlaying methods for allocating, releaseing and updating a memory
+slot. This implementation is going to be used by address allocator
+component of vm-allocator, its main goal is to make address slots
+allocation easier to track and maintain.
+
+```rust
+pub struct IntervalTree {
+    pub(crate) root: Option<InnerNode>,
```

Comment:
Do we need to have the `pub(crate)` visibility here? Is it possible to completely hide away this implementation detail?